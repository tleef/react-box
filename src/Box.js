import styled from 'styled-components'
import PropTypes from 'prop-types'
import paramCase from 'param-case'

// TODO replace with better prop-type definitions
export const propTypes = {
  alignContent: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
  alignItems: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
  alignSelf: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
  border: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
  borderBottom: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
  borderLeft: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
  borderRight: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
  borderTop: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
  bottom: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
  boxShadow: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
  boxSizing: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
  display: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
  flex: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
  flexBasis: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
  flexDirection: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
  flexFlow: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
  flexGrow: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
  flexShrink: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
  flexWrap: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
  height: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
  justifyContent: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
  left: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
  margin: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
  marginBottom: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
  marginLeft: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
  marginRight: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
  marginTop: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
  maxHeight: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
  maxWidth: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
  minHeight: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
  minWidth: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
  order: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
  overflow: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
  overflowX: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
  overflowY: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
  padding: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
  paddingBottom: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
  paddingLeft: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
  paddingRight: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
  paddingTop: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
  right: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
  top: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
  visibility: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
  width: PropTypes.oneOfType([PropTypes.number, PropTypes.string])
}

export const Box = styled.div`
  ${(props) => Object.keys(propTypes)
    .filter((k) => props[k])
    .map((k) => `${paramCase(k)}: ${props[k]};`)
    .join('')}
`

Box.displayName = 'Box'
Box.propTypes = propTypes
Box.defaultProps = {
  display: 'flex',
  boxSizing: 'border-box'
}

export default Box
