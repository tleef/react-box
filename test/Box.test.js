import React from 'react'
import renderer from 'react-test-renderer'
import 'jest-styled-components'
import Box from '../src/Box'

test('Box renders correctly', () => {
  const tree = renderer.create(<Box />).toJSON()

  expect(tree).toMatchSnapshot()
})