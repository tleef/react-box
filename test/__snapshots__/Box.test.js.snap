// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Box renders correctly 1`] = `
.c0 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

<div
  className="c0"
  display="flex"
/>
`;
